source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -147,18 +152,19 @@ SplitRequestPtr SimpleRequest::create(Router& router,
   std::unique_ptr<SimpleRequest> request_ptr{
       new SimpleRequest(callbacks, command_stats, time_source, latency_in_micros)};
 
-  auto conn_pool = router.upstreamPool(incoming_request->asArray()[1].asString());
-  if (conn_pool) {
-    request_ptr->conn_pool_ = conn_pool;
-    request_ptr->handle_ = conn_pool->makeRequest(incoming_request->asArray()[1].asString(),
-                                                  *incoming_request, *request_ptr);
+  std::string& key = incoming_request->asArray()[1].asString();
+  auto route = router.upstreamPool(key);
+  if (route) {
+    request_ptr->conn_pool_ = route->upstream();
+    request_ptr->handle_ = route->upstream()->makeRequest(key, *incoming_request, *request_ptr);
   }
 
   if (!request_ptr->handle_) {
     callbacks.onResponse(Utility::makeError(Response::get().NoUpstreamHost));
     return nullptr;
   }
 
+  mirrorRequest(route, incoming_request->asArray()[0].asString(), key, *incoming_request);
```

Comment:
It seems not optimal that we explicitly have to call `mirrorRequest` everywhere, and in general there is a lot of code duplication in all of these paths. Is there any way to consolidate at least the mirror flow?

Reply:
done